src/libostree/ostree-repo-commit.c

```diff
@@ -2557,6 +2571,7 @@ write_directory_content_to_mtree_internal (OstreeRepo                  *self,
   g_autoptr(GFileInfo) modified_info = NULL;
   OstreeRepoCommitFilterResult filter_result =
     _ostree_repo_commit_modifier_apply (self, modifier, child_relpath, child_info, &modified_info);
+  const gboolean child_info_was_modified = (child_info != modified_info);
```

Comment:
I think my main problem is this - if we have say a commit filter active and that filter just wants to change *some* paths like gnome-continuous does with suid binaries, this will still forcibly re-checksum everything since the info will be different.

I think we need to compare it structurally the same way we do xattrs.